From b98179f7a87f5a22225d3c742dc5a0773d0cd5ed Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Tue, 4 Mar 2014 12:37:48 +0800
Subject: [PATCH 0297/1074] ENGR00301635 ARM: dts: imx6sx: add thermal monitor
 support

add thermal monitor support for i.MX6SX.

Signed-off-by: Anson Huang <b20788@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/imx6sx.dtsi |   19 +++++++++++++++++++
 1 files changed, 19 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx.dtsi b/arch/arm/boot/dts/imx6sx.dtsi
index 59f9f87..69799ca 100644
--- a/arch/arm/boot/dts/imx6sx.dtsi
+++ b/arch/arm/boot/dts/imx6sx.dtsi
@@ -160,6 +160,14 @@
 				interrupts = <0 49 0x04 0 54 0x04 0 127 0x04>;
 			};
 
+			tempmon: tempmon {
+				compatible = "fsl,imx6sx-tempmon", "fsl,imx6q-tempmon";
+				interrupts = <0 49 0x04>;
+				fsl,tempmon = <&anatop>;
+				fsl,tempmon-data = <&ocotp>;
+				clocks = <&clks IMX6SX_CLK_PLL3_USB_OTG>;
+			};
+
 			flexcan1: can@02090000 {
 				compatible = "fsl,imx6q-flexcan";
 				reg = <0x02090000 0x4000>;
@@ -458,6 +466,17 @@
 				reg = <0x021b0000 0x4000>;
 			};
 
+			ocotp: ocotp-ctrl@021bc000 {
+				compatible = "syscon";
+				reg = <0x021bc000 0x4000>;
+			};
+
+			ocotp-fuse@021bc000 {
+				compatible = "fsl,imx6sx-ocotp", "fsl,imx6q-ocotp";
+				reg = <0x021bc000 0x4000>;
+				clocks = <&clks IMX6SX_CLK_OCOTP>;
+			};
+
 			qspi2: qspi@021e4000 {
 				#address-cells = <1>;
 				#size-cells = <0>;
-- 
1.7.5.4

